src/segments/typing.py

```diff
@@ -592,3 +592,49 @@ class SegmentsDatasetCategory(BaseModel):
     class Config:
         allow_population_by_field_name = True
         extra = "allow"
+
+
+####################
```

Comment:
Can we re-use the regular types? E.g. having both `Sample` and `ParsedSample` feels redundant and will probably confuse users.